client-js/src/client/actor-request-factory.js

```diff
@@ -20,162 +20,173 @@
-
-        let result = this._query(target);
-        return result;
-    }
-
-    /**
-     * Creates a Command from the given command message.
-     *
-     * @param message the command message, represented as a {@link TypedMessage}
-     * @returns a {@link TypedMessage} of the built command
-     */
-    command(message) {
-        let id = ActorRequestFactory._newCommandId();
-        let messageAny = message.toAny();
-        let context = this._commandContext();
-
-        let result = new command.Command();
-        result.setId(id);
-        result.setMessage(messageAny);
-        result.setContext(context);
-
-        let typedResult = new TypedMessage(result, COMMAND_MESSAGE_TYPE);
-        return typedResult;
-    }
-
-    _query(target) {
-        let id = ActorRequestFactory._newQueryId();
-        let actorContext = this._actorContext();
-
-        let result = new query.Query();
-        result.setId(id);
-        result.setTarget(target);
-        result.setContext(actorContext);
-
-        let typedResult = new TypedMessage(result, QUERY_MESSAGE_TYPE);
-        return typedResult;
-    }
-
-    _actorContext() {
-        let result = new actorContext.ActorContext();
-        result.setActor(this._actor);
-        let seconds = new Date().getUTCSeconds();
-        let time = new timestamp.Timestamp();
-        time.setSeconds(seconds);
-        result.setTimestamp(time);
-        result.setZoneOffset(ActorRequestFactory._zoneOffset());
-        return result;
-    }
-
-    _commandContext() {
-        let result = new command.CommandContext();
-        let actorContext = this._actorContext();
-        result.setActorContext(actorContext);
-        return result;
-    }
-
-    static _zoneOffset() {
-        let timeOptions = Intl.DateTimeFormat().resolvedOptions();
-        let zoneId = new time.ZoneId();
-        zoneId.setValue(timeOptions.timeZone);
-
-        let zoneOffset = ActorRequestFactory._zoneOffsetSeconds();
-
-        let result = new time.ZoneOffset();
-        result.setAmountSeconds(zoneOffset);
-        return result;
-    }
-
-    static _zoneOffsetSeconds() {
-        return new Date().getTimezoneOffset() * 60;
-    }
-
-    static _newQueryId() {
-        let result = new query.QueryId();
-        result.setValue("q-" + uuid.v4());
-        return result;
-    }
-
-    static _newCommandId() {
-        let result = new command.CommandId();
-        result.setUuid(uuid.v4());
-        return result;
-    }
+  /**
+   * Creates a new instance of ActorRequestFactory for the given actor.
+   *
+   * @param {!string} actor a string identifier of an actor
+   */
+  constructor(actor) {
+    this._actor = new UserId();
+    this._actor.setValue(actor);
+  }
+
+  /**
+   * Creates a Query targeting all the instances of the given type.
+   *
+   * @param {!TypeUrl} typeUrl a type URL of the target type
+   * @return {TypedMessage<Query>} a typed message of the Spine Query
+   */
+  newQueryForAll(typeUrl) {
```

Comment:
Can we please mimic the `QueryFactory` and `QueryBuilder` API approach (naming etc)?